src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.quarto.component.spec.ts

```diff
@@ -423,12 +422,95 @@ describe('OnlineGameWrapperComponent of Quarto:', () => {
+            await doMove(winningMove, true);
+
+            // then the game should be a victory
+            expect(wrapper.gameComponent.rules.node.move.get()).toEqual(winningMove);
+            expect(partDAO.update).toHaveBeenCalledOnceWith('joinerId', {
+                lastUpdate: {
+                    index: 6,
+                    player: observerRole.value,
+                },
+                listMoves: [move0, move1, move2, move3, winningMove].map(QuartoMove.encoder.encodeNumber),
+                turn: 5,
+                // remainingTimes are not present on the first move of a current board
+                request: null,
+                lastMoveTime: firebase.firestore.FieldValue.serverTimestamp(),
+                winner: 'creator',
+                loser: 'firstCandidate',
+                result: MGPResult.VICTORY.value,
+            });
+            componentTestUtils.expectElementToExist('#youWonIndicator');
+            expectGameToBeOver();
+        }));
+        it('Draw move from player should notifyDraw', fakeAsync(async() => {
+            //  Given a board on which user can draw
+            const moves: QuartoMove[] = [
+                new QuartoMove(0, 0, QuartoPiece.AAAB),
+                new QuartoMove(0, 1, QuartoPiece.AABA),
+                new QuartoMove(0, 2, QuartoPiece.BBAA),
+                new QuartoMove(0, 3, QuartoPiece.ABAA),
+
+                new QuartoMove(1, 0, QuartoPiece.ABAB),
+                new QuartoMove(1, 1, QuartoPiece.ABBA),
+                new QuartoMove(1, 2, QuartoPiece.BABB),
+                new QuartoMove(1, 3, QuartoPiece.BAAA),
+
+                new QuartoMove(2, 0, QuartoPiece.BAAB),
+                new QuartoMove(2, 1, QuartoPiece.BABA),
+                new QuartoMove(2, 2, QuartoPiece.ABBB),
+                new QuartoMove(2, 3, QuartoPiece.AABB),
+
+                new QuartoMove(3, 0, QuartoPiece.BBBA),
+                new QuartoMove(3, 1, QuartoPiece.BBAB),
+                new QuartoMove(3, 2, QuartoPiece.BBBB),
+            ];
+            await prepareBoard(moves, Player.ONE);
+            componentTestUtils.expectElementNotToExist('#winnerIndicator');
+
+            // when doing winning move
+            spyOn(partDAO, 'update').and.callThrough();
+            const drawingMove: QuartoMove = new QuartoMove(3, 3, QuartoPiece.NONE);
+            await doMove(drawingMove, true);
+
+            // then the game should be a victory
+            expect(wrapper.gameComponent.rules.node.move.get()).toEqual(drawingMove);
+            const listMoves: QuartoMove[] = moves.concat(drawingMove);
+            expect(partDAO.update).toHaveBeenCalledOnceWith('joinerId', {
+                lastUpdate: {
+                    index: 17,
+                    player: Player.ONE.value,
+                },
+                listMoves: listMoves.map(QuartoMove.encoder.encodeNumber),
+                turn: 16,
+                // TODO: investigate on why remainingTimes is not changed
```

Comment:
TODO introduced

Reply:
Non il est pas nouveau ! Je peux m'en charger dans j'investiguation sur les take back !

Reply:
ok! you can resoudrez